server/ServerApp.cpp

```diff
@@ -1763,7 +1771,8 @@ int ServerApp::AddPlayerIntoGame(const PlayerConnectionPtr& player_connection) {
     m_player_empire_ids[player_connection->PlayerID()] = empire_id;
 
     const OrderSet dummy;
-    const OrderSet& orders = orders_it->second.second ? *orders_it->second.second : dummy;
+    const OrderSet& orders = static_cast<bool>(orders_it->second.second) && static_cast<bool>(orders_it->second.second->m_orders) ? *(orders_it->second.second->m_orders) : dummy;
```

Comment:
why the extra explicit casting? should be implicitly if in a boolean test...

Reply:
Fixed.